nonblock-rpc/rust/core/src/router.rs

```diff
@@ -0,0 +1,95 @@
+use std::collections::HashMap;
+
+use serde::Serialize;
+
+pub struct Router {
+    pub routes: HashMap<String, Box<dyn Handler + 'static + Send + Sync>>,
+}
+
+impl Router {
+    pub fn new() -> Self {
+        Router {
+            routes: HashMap::new(),
+        }
+    }
+
+    pub fn add(mut self, name: &str, handler: impl Handler + 'static + Send + Sync) -> Self {
+        self.routes.insert(name.to_string(), Box::new(handler));
+        self
+    }
+
+    pub fn call(&self, name: &str, args: Vec<String>) -> String {
+        match self.routes.get(name) {
+            Some(handler) => handler.call(args),
+            None => "Unknown function".to_string(),
+        }
+    }
+}
+
+#[derive(Serialize)]
+pub struct Response<T> {
+    pub result: Option<T>,
+    pub error: Option<String>,
+    pub is_success: bool,
+}
+
+const CATASTROPHIC_ERROR: &str = "{\"result\": null, \"error\": \"CATASTROPHIC_ERROR: Failed to serialize response\", \"is_success\": false}";
```

Comment:
‘CATASTROPHIC_ERROR’ sounds pretty intense 😃 
Is there a specific reason for using such a strong term?

By the way, I think this might be something that could be serialized into JSON.